File: training-data/dart/matrix_rotate_90.dart
List<List<T>> rotateClockwise<T>(List<List<T>> m) => [
      for (var c = 0; c < m.first.length; c++) [for (var r = m.length - 1; r >= 0; r--) m[r][c]]
    ];

void main() {
  rotateClockwise([
    [1, 2, 3],
    [4, 5, 6],
    [7, 8, 9],
  ]).forEach(print);
}

File: training-data/swift/matrix_rotate_90.swift
func rotateClockwise<T>(_ m: [[T]]) -> [[T]] {
    guard let first = m.first else { return [] }
    return first.indices.map { c in m.reversed().map { $0[c] } }
}

let rotated = rotateClockwise([[1, 2, 3], [4, 5, 6], [7, 8, 9]])
rotated.forEach { print($0) }
